packages/sonarwhal/src/lib/utils/resource-loader.ts

```diff
@@ -280,3 +284,65 @@ export const loadFormatter = (formatterId: string): IFormatter => {
 export const loadParser = (parserId: string): Parser => {
     return loadResource(parserId, TYPE.parser);
 };
+
+const searchNpmPackages = (searchTerm: string): Promise<Array<NpmPackage>> => {
+    return new Promise((resolve, reject) => {
+        const results = [];
+
+        const searchOptions = {
+            description: true,
+            excluded: [],
+            include: [searchTerm],
+            limit: 1000,
+            staleness: 900,
+            unicode: false
+        };
+
+        esearch(searchOptions)
+            .on('data', (data) => {
+                results.push(data);
+            })
+            .on('error', (err) => {
+                reject(err);
+            })
+            .on('end', () => {
+                resolve(results);
+            });
+    });
+};
+
+const loadNpm = () => {
+    return npmLoadAsync({ loaded: false });
+};
+
+export const getNpmPackages = async (searchTerm: string = 'sonarwhal'): Promise<Array<NpmPackage>> => {
+    await loadNpm();
+
+    return searchNpmPackages(searchTerm);
+};
+
+const filterPackages = (packages: Array<NpmPackage>, initTerm: string) => {
+    return packages.filter((pkg) => {
+        return pkg.name.startsWith(initTerm);
+    });
+};
+
+export const getExternalRulesFromNpm = async () => {
+    const rules = await getNpmPackages('sonarwhal-rule');
+
+    /*
+     * We need to filter the results because the search can
+     * include other packages that doesn't start with `sonarwhal-rule`.
+     */
+    return filterPackages(rules, 'sonarwhal-rule');
+};
+
+export const getCoreRulesFromNpm = async () => {
```

Comment:
@alrra are we going to separate also the formatters, etc.? Right now?

If so then maybe we should have a `getCorePackages` that accepts a filter and build `getCore{Rule|Parser|Formatter|Connector}sFromNpm` on top of it?

Reply:
> are we going to separate also the formatters, etc.? Right now?

I think we should do it.